src/GitHub.VisualStudio/Commands/OpenFileInSolutionCommand.cs

```diff
@@ -0,0 +1,157 @@
+{
+    [Export(typeof(IOpenFileInSolutionCommand))]
+    public class OpenFileInSolutionCommand : VsCommand, IOpenFileInSolutionCommand
+    {
+        /// <summary>
+        /// Gets the GUID of the group the command belongs to.
+        /// </summary>
+        public static readonly Guid CommandSet = Guids.guidContextMenuSet;
+
+        /// <summary>
+        /// Gets the numeric identifier of the command.
+        /// </summary>
+        public const int CommandId = PkgCmdIDList.openFileInSolutionCommand;
+
+        readonly IGitHubServiceProvider serviceProvider;
+        readonly Lazy<IVsTextManager> textManager;
+        readonly Lazy<IVsEditorAdaptersFactoryService> editorAdapter;
+        readonly Lazy<IPullRequestSessionManager> sessionManager;
+        readonly Lazy<IPullRequestEditorService> pullRequestEditorService;
+        readonly Lazy<IStatusBarNotificationService> statusBar;
+        readonly Lazy<IUsageTracker> usageTracker;
+
+        [ImportingConstructor]
+        public OpenFileInSolutionCommand(
+            IGitHubServiceProvider serviceProvider,
+            Lazy<IVsEditorAdaptersFactoryService> editorAdapter,
+            Lazy<IPullRequestSessionManager> sessionManager,
+            Lazy<IPullRequestEditorService> pullRequestEditorService,
+            Lazy<IStatusBarNotificationService> statusBar,
+            Lazy<IUsageTracker> usageTracker) : base(CommandSet, CommandId)
+        {
+            textManager = new Lazy<IVsTextManager>(() => serviceProvider.GetService<SVsTextManager, IVsTextManager>());
+            this.serviceProvider = serviceProvider;
+            this.editorAdapter = editorAdapter;
+            this.sessionManager = sessionManager;
+            this.pullRequestEditorService = pullRequestEditorService;
+            this.statusBar = statusBar;
+            this.usageTracker = usageTracker;
+        }
+
+        public override async Task Execute()
+        {
+            try
+            {
+                var sourceView = FindActiveView();
+                if (sourceView == null)
+                {
+                    ShowErrorInStatusBar("Couldn't find active source view");
+                    return;
+                }
+
+                var textView = editorAdapter.Value.GetWpfTextView(sourceView);
+
+                var info = sessionManager.Value.GetTextBufferInfo(textView.TextBuffer);
+                if (info != null)
+                {
+                    if (!info.Session.IsCheckedOut)
+                    {
+                        ShowInfoMessage(App.Resources.NavigateToEditorNotCheckedOutInfoMessage);
+                        return;
+                    }
+
+                    // Navigate from PR file to solution
+                    await usageTracker.Value.IncrementCounter(x => x.NumberOfPRDetailsNavigateToEditor);
+
+                    await pullRequestEditorService.Value.OpenFile(info.Session, info.RelativePath, true);
+                    var fileView = FindActiveView();
```

Comment:
I think we need a delay here. Sometimes it doesn't pick up the destination file. :-(